enterprise/server/api/api_server.go

```diff
@@ -571,11 +572,20 @@ func (s *APIServer) Run(ctx context.Context, req *apipb.RunRequest) (*apipb.RunR
 		})
 	}
 
+	patches := make([][]byte, len(req.GetPatches()))
+	for i, p := range req.GetPatches() {
+		data, err := base64.StdEncoding.DecodeString(p)
+		if err != nil {
+			return nil, status.InvalidArgumentErrorf("invalid patch: %s", err)
+		}
+		patches[i] = data
+	}
 	rsp, err := r.Run(ctx, &rnpb.RunRequest{
 		GitRepo: &gitpb.GitRepo{RepoUrl: req.GetRepo()},
 		RepoState: &gitpb.RepoState{
 			CommitSha: req.GetCommitSha(),
 			Branch:    req.GetBranch(),
+			Patch:     patches,
```

Comment:
(see other comment re changing `string` => `bytes`)

```suggestion
	rsp, err := r.Run(ctx, &rnpb.RunRequest{
		GitRepo: &gitpb.GitRepo{RepoUrl: req.GetRepo()},
		RepoState: &gitpb.RepoState{
			CommitSha: req.GetCommitSha(),
			Branch:    req.GetBranch(),
			Patch:     req.GetPatches(),
```